packages/search/graphql/resolvers/_queries/search.js

```diff
@@ -0,0 +1,444 @@
+
+  return indices.list.filter(indicesFilter)
+}
+
+const search = async (__, args, context) => {
+  const { user, elastic, t } = context
+  const {
+    after, before, skipLoadRelatedDocs = false, scheduledAt
+  } = args
+
+  const options = after
+    ? { ...args, ...parseOptions(after) }
+    : before
+      ? { ...args, ...parseOptions(before) }
+      : args
+
+  const {
+    search,
+    filter: filterObj = { },
+    filters: filterArray = [],
+    sort = {
+      key: 'relevance',
+      direction: 'DESC'
+    },
+    first: _first = 40,
+    from = 0
+  } = options
+
+  debug('options', JSON.stringify(options))
+
+  Object.keys(filterObj).forEach(key => {
+    filterArray.push({
+      key,
+      value: filterObj[key]
+    })
+  })
+
+  const filter = reduceFilters(filterArray)
+
+  debug('filter', JSON.stringify(filter))
+
+  const first = getFirst(_first, filter, user)
+
+  const indicesList = getIndicesList(filter)
+  const query = {
+    index: indicesList.map(({ name }) => getIndexAlias(name, 'read')),
+    from,
+    size: first,
+    body: createQuery(search, filter, sort, indicesList, user, scheduledAt)
+  }
+
+  debug('ES query', JSON.stringify(query))
+  const result = await elastic.search(query)
+  // debug('result: %O', result)
+
+  const hasNextPage = first > 0 && result.hits.total > from + first
+  const hasPreviousPage = from > 0
+
+  const response = {
+    nodes: result.hits.hits.map(mapHit),
+    aggregations: mapAggregations(result, t),
+    totalCount: result.hits.total,
+    pageInfo: {
+      hasNextPage,
+      endCursor: hasNextPage
+        ? stringifyOptions(cleanOptions({
+          ...options,
+          filters: filterArray,
+          first,
+          from: from + first
+        }))
+        : null,
+      hasPreviousPage,
+      startCursor: hasPreviousPage
+        ? stringifyOptions(cleanOptions({
+          ...options,
+          filters: filterArray,
+          first,
+          from: from - first
+        }))
+        : null
+    }
+  }
+
+  if (!skipLoadRelatedDocs && (!filter.type || filter.type === 'Document')) {
+    await addRelatedDocs({
+      connection: response,
+      context
+    })
+  }
+
+  if (!skipLoadRelatedDocs && SEARCH_TRACK) {
+    try {
+      const took = result.took
+      const total = result.hits.total
+      const hits = result.hits.hits
+        .map(hit => _.omit(hit, '_source'))
+
+      const filters = options.filters
+        ? options.filters.map(filter => {
+          if (typeof filter.value !== 'string') {
+            filter.value = JSON.stringify(filter.value)
+          }
+
+          return filter
+        })
+        : []
+
+      await elastic.index({
+        index: getIndexAlias('searches'),
+        type: 'Search',
+        body: {
+          took,
+          options: Object.assign({}, options, { filters }),
+          query,
+          total,
+          hits,
+          date: new Date(),
+          user: {
+            id: (user && user.id) || 'anon'
```

Comment:
This is not permissible, this becomes personal data if you connect it here. Please remove the user context.